tests/test_models_variational.py

```diff
@@ -268,11 +268,14 @@ def test_qaoa_errors(backend):
 
 @pytest.mark.parametrize(test_names, test_values)
 def test_qaoa_optimization(backend, method, options, dense, filename):
-    if method == "sgd" and backend.name != "tensorflow":
+    if (method == "sgd") and (backend.name not in ["tensorflow", "pytorch"]):
         pytest.skip("Skipping SGD test for unsupported backend.")
+    if method != "sgd" and backend.name in ("tensorflow", "pytorch"):
+        pytest.skip("Skipping scipy optimizers for tensorflow and pytorch.")
```

Comment:
same as above

Reply:
also here we are using all backends, check the second condition